src/xo-app/sr/tab-disks.js

```diff
@@ -89,13 +51,61 @@ const COLUMNS = [
     itemRenderer: vdi => formatSize(vdi.size),
     sortCriteria: vdi => vdi.size,
   },
+  {
+    name: _('vdiVm'),
+    component: connectStore(() => {
+      const getVbds = createGetObjectsOfType('VBD').pick(
+        (_, props) => props.item.$VBDs
+      )
+      const getVmIds = createSelector(getVbds, vbds => map(vbds, 'VM'))
+      const getVms = createGetObjectsOfType('VM').pick(getVmIds)
+
+      return (state, props) => ({
+        vms: getVms(state, props),
+        vbds: getVbds(state, props),
+      })
+    })(({ vbds, vms }) => {
+      if (isEmpty(vms)) {
+        return null
+      }
+
+      return (
+        <Container>
+          {map(vms, vm => {
+            const vmId = vm.id
+            const item = renderXoItem(vm)
+            const vbd = find(vbds, { VM: vmId })
```

Comment:
I can't reply to an outdated comment :weary:
So, it's true, you already have a small component, keep it like this. What's bothering me is this `find` (1) for each VDI (2), on each render (3) and for each VM (4). That's a lot of loops :smile:. So you could simply loop over the VBDs instead of the VMs. And in order to find the corresponding VM for each VBD, you can do `vms[vbd.VM]`.